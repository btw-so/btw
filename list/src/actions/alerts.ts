import React from 'react';
import { uuid } from '@gilbarbara/helpers';
import { createAction } from '@reduxjs/toolkit';

import { actionPayload } from 'modules/helpers';

import { ActionTypes } from 'literals';

import { ShowAlertOptions } from 'types';

export const hideAlert = createAction<string>(ActionTypes.HIDE_ALERT);

export const showAlert = createAction(
  ActionTypes.SHOW_ALERT,
  (message: React.ReactNode, options: ShowAlertOptions) => {
    const timeout = options.variant === 'danger' ? 0 : 5;

    return actionPayload({
      id: options.id || uuid(),
      icon: options.icon || 'dot-circle-o',
      message,
      position: options.position || 'bottom-right',
      variant: options.variant || 'dark',
      timeout: typeof options.timeout === 'number' ? options.timeout : timeout,
    });
  },
);
