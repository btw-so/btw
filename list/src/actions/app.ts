import { createAction } from '@reduxjs/toolkit';

import { ActionTypes } from 'literals';

import { SetAppOOptions } from 'types';

export const setAppOptions = createAction<SetAppOOptions>(ActionTypes.SET_APP_OPTIONS);
