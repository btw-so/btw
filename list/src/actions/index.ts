export * from './alerts';
export * from './app';
export * from './github';
export * from './user';
export * from './list';
export * from './notes';