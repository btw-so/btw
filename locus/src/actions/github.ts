import { createAction } from '@reduxjs/toolkit';

import { actionPayload } from 'modules/helpers';

import { ActionTypes } from 'literals';

export interface GetReposSuccessMeta {
  cached: boolean;
  query: string;
  updatedAt: number;
}

export const getRepos = createAction<string>(ActionTypes.GITHUB_GET_REPOS_REQUEST);
export const getReposSuccess = createAction(
  ActionTypes.GITHUB_GET_REPOS_SUCCESS,
  (payload: Record<string, any>[], meta: GetReposSuccessMeta) => actionPayload(payload, meta),
);
export const getReposFailure = createAction(
  ActionTypes.GITHUB_GET_REPOS_FAILURE,
  (payload: string, query: string) => actionPayload(payload, { query }),
);
