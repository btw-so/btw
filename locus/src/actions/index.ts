export * from './alerts';
export * from './app';
export * from './github';
export * from './user';
export * from './list';
export * from './notes';
export * from './files';
export * from './intelligence';