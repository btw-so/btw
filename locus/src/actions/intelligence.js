
import { createAction } from "@reduxjs/toolkit";

import { ActionTypes } from "literals";

export const saveIntelligenceApiKeys = createAction(
  ActionTypes.SAVE_INTELLIGENCE_API_KEYS,
  (apiKeys) => ({
    payload: { apiKeys },
  })
);

export const savePreferredTabs = createAction(
  ActionTypes.SAVE_PREFERRED_TABS,
  (preferredTabs) => ({
    payload: { preferredTabs },
  })
);
