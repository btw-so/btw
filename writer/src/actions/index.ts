export * from './alerts';
export * from './app';
export * from './github';
export * from './user';
export * from './notes';
export * from './list';