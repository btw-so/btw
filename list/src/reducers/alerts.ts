import { createReducer } from '@reduxjs/toolkit';

import { hideAlert, showAlert } from 'actions';

import { AlertsState } from 'types';

export const alertsState: AlertsState = {
  data: [],
};

export default {
  alerts: createReducer<AlertsState>(alertsState, builder => {
    builder
      .addCase(hideAlert, (draft, { payload }) => {
        draft.data = draft.data.filter(d => d.id !== payload);
      })
      .addCase(showAlert, (draft, { payload }) => {
        draft.data.push(payload);
      });
  }),
};
