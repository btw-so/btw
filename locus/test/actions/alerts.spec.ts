import { hideAlert, showAlert } from 'actions/alerts';

describe('actions/alerts', () => {
  describe('showAlert', () => {
    it('should return an action with variant `error`', () => {
      expect(showAlert('Alright!', { id: 'test', variant: 'danger' })).toMatchSnapshot();
    });

    it('should return an action with variant `success`', () => {
      expect(
        showAlert('Alright!', { id: 'test', variant: 'success', timeout: 10 }),
      ).toMatchSnapshot();
    });
  });

  describe('hideAlert', () => {
    it('should return an action', () => {
      expect(hideAlert('test')).toMatchSnapshot();
    });
  });
});
