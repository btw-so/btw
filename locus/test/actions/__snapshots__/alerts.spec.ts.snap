// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions/alerts hideAlert should return an action 1`] = `
Object {
  "payload": "test",
  "type": "HIDE_ALERT",
}
`;

exports[`actions/alerts showAlert should return an action with variant \`error\` 1`] = `
Object {
  "meta": undefined,
  "payload": Object {
    "icon": "dot-circle-o",
    "id": "test",
    "message": "Alright!",
    "position": "bottom-right",
    "timeout": 0,
    "variant": "danger",
  },
  "type": "SHOW_ALERT",
}
`;

exports[`actions/alerts showAlert should return an action with variant \`success\` 1`] = `
Object {
  "meta": undefined,
  "payload": Object {
    "icon": "dot-circle-o",
    "id": "test",
    "message": "Alright!",
    "position": "bottom-right",
    "timeout": 10,
    "variant": "success",
  },
  "type": "SHOW_ALERT",
}
`;
